app/lib/backend/model_extensions/cinstance.rb

```diff
@@ -83,18 +56,6 @@ def preload_used_associations
       def set_application_id
         self.application_id ||= SecureRandom.hex(4)
       end
-
-      def update_provider_backend_service_if_user_key_changed
-        if previously_changed?(:user_key)
-          user_key_was, user_key = previous_changes[:user_key]
-
-          if user_account && user_account.provider? && user_key_was.present? && user_account.services.present?
-            ThreeScale::Core::Service.change_provider_key!(user_key_was, user_key)
```

Comment:
It looks to me this is legit call,

In pisoni master it is not marked as deprecate https://github.com/3scale/pisoni/blob/9904bce5dfb482a49a49fb5684f6262af7591ada/lib/3scale/core/service.rb#L26-L38

Or should it? Ref: https://issues.redhat.com/browse/THREESCALE-5152

Reply:
Makes sense 👍

Reply:
This should not be changed

Reply:
u r right! 🙏 💪